### ISSUE TITLE
gatsby-plugin-google-analytics: `sendPageView` should be called via `requestAnimationFrame`, when available

### ISSUE THREAD

**Issue by d4h0** [gatsbyjs/gatsby#28591]
React-Helm [uses requestAnimationFrame to avoid FOUC](https://github.com/nfl/react-helmet/pull/307).

This lead to a [bug](https://github.com/gatsbyjs/gatsby/issues/9139) in `gatsby-plugin-google-analytics` where a page view was sent with the title of the previous page.

[A workaround](https://github.com/gatsbyjs/gatsby/pull/10917) was implemented to fix the problem:

```javascript
    if (`requestAnimationFrame` in window) {
      requestAnimationFrame(() => {
        requestAnimationFrame(sendPageView)
      })
    } else {
      // simulate 2 rAF calls
      setTimeout(sendPageView, 32)
    }
```

For some reason, `requestAnimationFrame` was removed later ([see here](https://github.com/gatsbyjs/gatsby/blob/305e7c57f9da71bc427cddf1e69ad3b42ee86afe/packages/gatsby-plugin-google-analytics/src/gatsby-browser.js#L24)):

```javascript
  // Minimum delay for reactHelmet's requestAnimationFrame
  const delay = Math.max(32, pluginOptions.pageTransitionDelay || 0)
  setTimeout(sendPageView, delay)
```

Basically, this code assumes that every display has a refresh rate of 60, which is not the case (the iPad Pro has a refresh rate of 120, for example).

Because React-Helm uses `requestAnimationFrame` when available (and this plugin does not), this means, that on devices with a higher refresh rate than 60, the wrong title  still could be sent.

See [MDN on requestAnimationFrame](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame):

> The number of callbacks is usually 60 times per second, but will generally match the display refresh rate in most web browsers as per W3C recommendation.

An easy fix would be, to restore the original implementation (see above)

**Comment by LekoArts** [comment]
We'd be happy to receive a PR to fix this. Thanks for the issue!

**Comment by d4h0** [comment]
@LekoArts: Sorry, I'm not using Google Analytics myself, and I'm behind with my own projects, so I won't create a PR myself (researching and reporting this bug used up all my available time for this).

**Comment by d4h0** [comment]
> If we missed this issue or if you want to keep it open, please reply here.

Looks like nobody is interested in fixing this, but this issue is still about a bug that should get fixed at some point, so it should be kept open, IMO.

**Comment by seanflahertyT4** [comment]
Hi, is this still an issue?  I am experiencing incorrect (unset) title names on google analytics.  If so, I can do the PR.
